src/components/route-button-container.tsx

```diff
@@ -12,38 +19,38 @@ const styles = StyleSheet.create({
 	},
 });
 
-interface Props {
-	/** Function to start and stop route tracking */
-	changeTracking: () => void;
-	/** Function to toggle show route state true or false */
-	changeShowRoute: () => void;
-	/** State to determine whether route is visible or not */
-	showRoute: boolean;
-	/** State to determine whether route tracking has started or not */
-	isTracking: boolean;
-}
-
 /**
  * Route button container component which contains buttons to
  * start / end route tracking and to toggle route visibility on / off.
  * Just preliminary styling and location on a screen.
- * @returns a tree of React elements
  */
-const RouteButtonContainer = ({
-	changeTracking,
-	changeShowRoute,
-	showRoute,
-	isTracking,
-}: Props): JSX.Element => {
+const RouteButtonContainer = (): JSX.Element => {
+	const userId: string | null = useTypedSelector((state) => state.user.userId);
+	const routeInfo = useTypedSelector((state) => state.route);
+	const dispatch = useTypedDispatch();
+
+	const changeRouteVisibility = () => {
+		dispatch(changeShowRoute(routeInfo));
+	};
+
+	const changeTracking = () => {
+		if (routeInfo.active) {
+			dispatch(deactivateRoute(routeInfo.routeId));
+			dispatch(resetWaypoints());
+		} else {
+			dispatch(startRoute(userId));
```

Comment:
Both the `deactivateRoute(…)` and `startRoute(…)` calls have type problems: they only accept `string` but are getting `string | null` passed in.